File: Ejemplos JavaScript/distribucionSort.js
// Ejemplo de distribución Sort
let numeros = [9, 8, 7, 6, 5, 4, 3, 2, 1];
let numerosOrdenados = new Array(9).fill(0);
let numerosOrdenados2 = new Array(9).fill(0);
let pivote = numeros[0];
let pivote2 = numeros[8];
let contadorPivote = 0;
let contadorPivote2 = 0;

// Ordenar números
for (let i = 0; i < 9; i++) {
  if (numeros[i] < pivote) {
    numerosOrdenados[contadorPivote] = numeros[i];
    contadorPivote++;
  }
  if (numeros[i] > pivote2) {
    numerosOrdenados2[contadorPivote2] = numeros[i];
    contadorPivote2++;
  }
}

// Mostrar números ordenados
console.log("Números ordenados:");
for (let i = 0; i < contadorPivote; i++) {
  process.stdout.write(numerosOrdenados[i] + " ");
}
for (let i = 0; i < contadorPivote2; i++) {
  process.stdout.write(numerosOrdenados2[i] + " ");
}
console.log();

File: ejemplosPython/distribucion_sort.py
# Ejemplo de distribución Sort
numeros = [9, 8, 7, 6, 5, 4, 3, 2, 1]
numeros_ordenados = [0] * 9
numeros_ordenados2 = [0] * 9
pivote = numeros[0]
pivote2 = numeros[8]
contador_pivote = 0
contador_pivote2 = 0

# Ordenar números
for i in range(9):
    if numeros[i] < pivote:
        numeros_ordenados[contador_pivote] = numeros[i]
        contador_pivote += 1
    if numeros[i] > pivote2:
        numeros_ordenados2[contador_pivote2] = numeros[i]
        contador_pivote2 += 1

# Mostrar números ordenados
print("Números ordenados:")
for i in range(contador_pivote):
    print(numeros_ordenados[i], end=" ")
for i in range(contador_pivote2):
    print(numeros_ordenados2[i], end=" ")
print()